protocol.go

```diff
@@ -11,7 +11,10 @@ import (
 // Listener is used to wrap an underlying listener,
 // whose connections may be using the HAProxy Proxy Protocol.
 // If the connection is using the protocol, the RemoteAddr() will return
-// the correct client address.
+// the correct client address. ReadHeaderTimeout will be applied to all
+// connections in order to prevent blocking operations. If no ReadHeaderTimeout
+// is set, a default of 200ms will be used. This can be disabled by setting the
+// timeout to < 0.
```

Comment:
I think we're missing a test for `ReadHeaderTimeout < 0`, right?

Reply:
Technically not as viewed from a coverage standpoint, but I can add a test for it. The reason I say not is because we'd expect the default to occur without setting deadlines, which is the read would hang forever.

Reply:
Added a test and cleaned up some lints for golangci-lint

Reply:
> Added a test and cleaned up some lints for golangci-lint

you included the same changes in #82, I'll close my PR